app/models/project.rb

```diff
@@ -705,6 +708,39 @@ def move_to_portfolio(file, name, kind)
     result
   end
 
+  def move_draft_to_portfolio
+    portfolio_tmp_dir = portfolio_temp_path
+    FileUtils.mkdir_p(portfolio_tmp_dir)
+    result = {}
+
+    # get draft learning summary task and continue if has_pdf
+    task = tasks.where(task_definition: unit.draft_task_definition_id).first
+    return false if task.nil?
+    return false if !task.has_pdf
+
+    # Extract to porfolio temp directory
+    zip_file = task.zip_file_path_for_done_task
+    if zip_file && File.exist?(zip_file)
+      Zip::File.open(zip_file) do |zip|
+        zip.glob("**/*").each do |entry|
+          next if entry.name_is_directory?
+          result[:name] = "DraftLearningSummaryReport.pdf"
```

Comment:
Why not move this in place as the learning summary report? No need for a separate "draft". That way it will already be the learning summary report when the portfolio is created. Otherwise you will need to copy the draft to real on creation of the portfolio.